src/models/generic/WorkflowModel.ts

```diff
@@ -1029,4 +1052,176 @@ export abstract class WorkflowModel extends ValidationBase implements Serializab
+        const destinations = this.gatherDestinations();
+
+        this.connections.forEach((connection) => {
+
+            if (connection.isVisible) {
+                const source = sources.find((item) => {
+                    return item.connectionId === connection.source.id;
+                });
+
+                const destination = destinations.find((item) => {
+                    return item.connectionId === connection.destination.id;
+                });
+
+                this.validateConnection(destination, source);
+            }
+        });
+
+        return new Promise((resolve) => resolve());
+    }
+
+    /**
+     * Validate connection between source and destination and sets correct validity state on destination
+     */
+    private validateConnection (destination: WorkflowOutputParameterModel | WorkflowStepInputModel,
+                        source: WorkflowInputParameterModel | WorkflowStepOutputModel, updateConnection = false) {
+
+        if (!source || !destination) {
+            return;
+        }
+
+        let isValid = true;
+
+        const destinationType = destination.type;
+        const sourceType = source.type;
+
+        // Check file types, true if at least one source file type is in destination file types
+        const checkFileTypes = () => {
+            if (source.fileTypes.length && !source.fileTypes
+                    .some(sourceType => !!destination.fileTypes
+                        .find((destType) => sourceType.toLowerCase() === destType.toLowerCase()))) {
+                return false;
+            } else {
+                return true;
+            }
+        };
+
+        // If type is the same (string -> string...) or type is array and items has the same type ([string] -> [string]...)
+        if (destinationType.type === sourceType.type && destinationType.items === sourceType.items) {
+
+            if (destinationType.type === "File" || destinationType.items === "File") {
+                isValid = checkFileTypes();
+            }
+
+        } else {
+
+            // If destination is array of items of type source (string -> [string]...)
+            if (destinationType.items === sourceType.type) {
+                isValid = checkFileTypes();
+            } else {
+                isValid = false;
+            }
+        }
+
+        // Whether to update connections or not (false when called from connectSource function)
+        if (updateConnection) {
+            this.connections.filter((c) =>
+                    c.isVisible && (c.destination.id === destination.connectionId && c.source.id === source.connectionId))
+                .forEach((c) => {
+                c.isValid = isValid;
+            });
+        }
+
+        if (!isValid) {
+            // Set warning
+            destination.updateValidity({[destination.loc + ".sources[" + source.sourceId + "]"]: {
+                message: `invalid connection (type or file types mismatch)`,
```

Comment:
This error can be much more descriptive. Form it at the moment you know what kind of validity it is (line 1118).

For type mismatches:
`Invalid connection. Connection type mismatch, attempting to connect "${source.type}" to "${destination.type}"`
 
For file type mismatches:
`Invalid connection. File type mismatch, connecting formats "${source.fileTypes}" to "${destinations.fileTypes}"`